tests/integrations/server/raft_client.rs

```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
     }
     pool.shutdown().wait().unwrap();
 }
+
+
+#[test]
+fn test_batch_raft_server_drop() {
+    #[derive(Clone)]
+    struct MockKvForRaft(Arc<AtomicUsize>);
+
+    impl MockKvService for MockKvForRaft {
+        fn raft(
```

Comment:
I think we don't need to implement this, just put logic into `batch_raft` is fine.